src/ssl.c

```diff
@@ -45356,9 +45357,13 @@ WOLFSSL_ASN1_TIME* wolfSSL_ASN1_TIME_adj(WOLFSSL_ASN1_TIME *s, time_t t,
         utc_hour = ts->tm_hour;
         utc_min  = ts->tm_min;
         utc_sec  = ts->tm_sec;
-        XSNPRINTF((char *)utc_str, sizeof(utc_str),
-                  "%02d%02d%02d%02d%02d%02dZ",
-                  utc_year, utc_mon, utc_day, utc_hour, utc_min, utc_sec);
+        sn_len = XSNPRINTF((char *)utc_str, sizeof(utc_str),
+                           "%02d%02d%02d%02d%02d%02dZ",
+                           utc_year, utc_mon, utc_day, utc_hour, utc_min,
+                           utc_sec);
+        if (sn_len > ASN_UTC_TIME_SIZE) {
+            return NULL;
```

Comment:
You will leak `s`. Please call `wolfSSL_ASN1_TIME_free(s);` Also would prefer `(int)sizeof(utc_str)` to prevent any future issue.

Why is the `XSNPRINTF` below also not checked? Perhaps `ASN_UTC_TIME_SIZE` needs a +1?

Reply:
1) Added free for s
2) Changed to (int)sizeof(..)
3) Because ASN_GENERALIZED_TIME_MAX is 68 (large enough to hold already, tool was not complaining)
--3.a) Added a check even though buffer large enough and no warning as a good practice!

Reply:
`snprintf` can also return a negative error. Consider also checking for `sn_len < 0`. This fix still doesn't make any sense to me, since snprintf will never return more than `sizeof(utc_str)`. Are you sure the solution isn't just increasing the size of `ASN_UTC_TIME_SIZE`?